### ISSUE TITLE
How to train the model by myself ?

### ISSUE THREAD

**Issue by thesby** [torrvision/crfasrnn#48]
I want to use the crfasrnn model to train on my data, but there is no network for training. Should I just add a data layer to provide label, and add a softmax layer to get loss ? If not, please give some  suggestions about it. Thank you.

**Comment by bittnt** [comment]
https://github.com/bittnt/caffe/tree/22babdb4a70eb5829c52fccc417d4b2f916d2748/examples/segmentationcrfasrnn

**Comment by westeast** [comment]
@thesby Have you trained your model by using your data?

**Comment by thesby** [comment]
@westeast Yes, I have trained one.

**Comment by chitrang6** [comment]
Hey I have a data set of cars and non cars image and using that I want to
detect cars on the highway. How to train using my dataset? Kindly help

**Comment by thesby** [comment]
So you want to do object detection or image segmentation ? 
If you just want to segment images, just use  image data layer to provide images, and add a crf layer as the paper did.

**Comment by marticsvk** [comment]
@thesby Hi thesby, i'm currently working on training my own data set based on this CNN-CRF model for image segmentation. Could you please write me your email so we can communicate there or directly send me a message at joannytrance@gmail.com, i'd appreciate your help.

**Comment by matthiasmace** [comment]
@bittnt 
Hi bittnt ! 
I want to train on my data (animal art representation). I followed the link mentionned above. It is pretty easy to understand but could you please explain the following in details : 

rgb-train"
label-train"
 rgb-val"
 label-val"

Many thanks !  By the way, it now works fine on my Macos 10.12 !

**Comment by bittnt** [comment]
These are the folder names of lmdbs. rgb-train is the lmdb folder for the input image data, while the label-train is for the label map.

You should be able to create lmdb following the caffe fashion.

**Comment by matthiasmace** [comment]
@bittnt 
OK, thanks.
But What do I need to put inside label-train please ? Do I follow the indications given in this link ?
http://host.robots.ox.ac.uk/pascal/VOC/voc2012/segexamples/index.html
And if yes, "object" or "class" ?
Thanks again

**Comment by bittnt** [comment]
``class``.

``object`` is for instance segmentation.
